### PR TITLE
chore: add deprecation

### PR DESCRIPTION
**Pull request by johanneswuerbach**
#### Description

Add deprecation notice. [[Rendered](https://github.com/johanneswuerbach/score-humanitec/blob/deprecation/README.md#-score-overview)].

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New chore (expected functionality to be implemented)

#### Checklist:

- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] I've signed off with an email address that matches the commit author.

### CODE REVIEW

**Comment by astromechza** [comment]
@johanneswuerbach can we update the code to log a message indicating the deprecation? I worry folks won't really see this update in the docs if they are already using score-humanitec in workflows.

**Review by sujaya-sys — APPROVED** [review]

File: .goreleaser.yaml [hunk 1]
```diff
@@ -42,3 +42,5 @@ brews:
     commit_author:
       name: rachfop
       email: prachford@icloud.com
+    custom_block: |
```
Comment (johanneswuerbach) [inline .goreleaser.yaml:45]: `custom_block` has described here https://goreleaser.com/customization/homebrew/

File: .goreleaser.yaml [hunk 2]
```diff
@@ -42,3 +42,5 @@ brews:
...
+      deprecate! date: "2024-05-08", because: "has been replaced by humctl score deploy. For more information visit https://developer.humanitec.com/platform-orchestrator/reference/cli-references/#score-integration."
```
Comment (johanneswuerbach) [inline .goreleaser.yaml:46]: `deprecate!` as described in https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae

Comment (astromechza) [reply .goreleaser.yaml:46]: that's very useful!

**Review by astromechza — APPROVED** [review]

**Pull request merged**